# Engine will crash if there are no "file" in protocols list
--- mozilla/chrome/src/nsChromeRegistry.cpp.orig	2007-10-03 07:09:45.000000000 +0300
+++ mozilla/chrome/src/nsChromeRegistry.cpp	2007-10-03 07:33:53.000000000 +0300
@@ -1108,20 +1108,24 @@ nsChromeRegistry::CheckForNewChrome()
   nsCOMPtr<nsIURI> manifestURI;
   rv = NS_NewURI(getter_AddRefs(manifestURI),
                  NS_LITERAL_CSTRING("resource:///chrome/app-chrome.manifest"));
 
   // this is the main manifest; if it doesn't exist we generate it from
   // installed-chrome.txt. When the build system learns about the new system,
   // this code can go away.
 
   nsCOMPtr<nsIFileURL> manifestFileURL (do_QueryInterface(manifestURI));
   NS_ASSERTION(manifestFileURL, "Not a nsIFileURL!");
+  if (!manifestFileURL) {
+    printf("Error: 'file' protocol disabled\n");
+    return NS_ERROR_FAILURE;
+  }
 
   nsCOMPtr<nsIFile> manifest;
   manifestFileURL->GetFile(getter_AddRefs(manifest));
   NS_ENSURE_TRUE(manifest, NS_ERROR_FAILURE);
 
   PRBool exists;
   rv = manifest->Exists(&exists);
   NS_ENSURE_SUCCESS(rv, rv);
 
 #ifdef DEBUG
